runtime/mux.go

```diff
@@ -75,6 +76,30 @@ type ServeMux struct {
 // ServeMuxOption is an option that can be given to a ServeMux on construction.
 type ServeMuxOption func(*ServeMux)
 
+// ForwardResponseRewriter is the signature of a function that is capable of rewriting messages
+// before the are forwarded in unary response, stream response, and error responses.
+type ForwardResponseRewriter func(ctx context.Context, response proto.Message) (interface{}, error)
+
+// WithForwardResponseRewriter returns a ServeMuxOption that allows for implementers to insert logic
+// that can rewrite the final response before it is forwarded.
+//
+// The response rewriter function is called during unary message forwarding, stream message
+// forwarding and when errors are being forwarded.
```

Comment:
Could you add a section to say that using this option to modify the response will likely make it incompatible with the schema generated by `protoc-gen-openapiv2` please?

Reply:
Done!